Sources/NIO/BlockingIOThreadPool.swift

```diff
@@ -153,6 +155,30 @@ public final class BlockingIOThreadPool {
     }
 }
 
+public extension BlockingIOThreadPool {
+    /// Runs the submitted closure if the thread pool is still active, otherwise fails the promise.
+    /// The closure will be run on the thread pool so can do blocking work.
+    ///
+    /// - parameters:
+    ///     - promise: The `EventLoopPromise` that will be fulfilled when the block finished running (or failed).
+    ///     - body: The closure which performs some blocking work to be done on the thread pool.
+    /// - returns: The `EventLoopFuture` of `promise` fulfilled with the result (or error) of the passed closure.
+    public func runIfActive<T>(promise: EventLoopPromise<T>, _ body: @escaping () throws -> T) -> EventLoopFuture<T> {
```

Comment:
@weissi @Lukasa sorry but I need to correct my previous suggestion here... whenever we take a `promise` we use `Void` as return type. So maybe we should do the same here as we ??

Reply:
Yeah, probably. It's likely to be non-obvious that the `Future` returned here is the one corresponding to the passed-in `promise`.

Reply:
@Lukasa and @normanmaurer can we please go back to taking the `EventLoop`? It's important that we return an `EventLoopFuture` as the next thing you want to do is clearly work with that file. The taking-the-promise approach is just a performance improvement if you might not care about the result of the operation (usually for `EventLoopFuture<Void>`s). For opening files however we absolutely do care about the return value. So we will need a promise&future here and I still think we should take the `EventLoop`

Reply:
@Lukasa & @normanmaurer updated this guy